packages/design-system/src/components/Table/Table.jsx

```diff
@@ -0,0 +1,151 @@
+  // https://reactjs.org/docs/hooks-effect.html
+  useEffect(() => {
+    // Set `tabIndex` to `0` if the table width is wider than the viewport and Table is set for horizontal scrolling.
+    if (scrollable) {
+      const { scrollWidth, clientWidth } = container.current;
+      const scrollableActive = scrollWidth > clientWidth;
+      setTabIndex(scrollableActive ? '0' : null);
+
+      const debouncedHandleResize = debounce(function handleResize() {
+        const { scrollWidth, clientWidth } = container.current;
+        const scrollableActive = scrollWidth > clientWidth;
+        setTabIndex(scrollableActive ? '0' : null);
+      }, 500);
+
+      // Create window resize event listener that calls the debouncedHandleResize function
+      window.addEventListener('resize', debouncedHandleResize);
+
+      // Specify how to clean up after this effect to avoid memory leak by removing establishd event listeners
+      return () => {
+        window.removeEventListener('resize', debouncedHandleResize);
+      };
+    }
+  });
+
+  const classes = classNames(
+    'ds-c-table',
+    striped ? 'ds-c-table--striped' : null,
+    responsiveTable ? `ds-c-table--stacked-${responsiveTable}` : null,
+    className
+  );
+
+  // `tabIndex` state is needed to make table container focusable and to display scrollable message when table width exceeds viewport.
+  // Set attribute `tabIndex = 0` to make table container focusable to enable keyboard support of using the arrow keys.
+  // Also, provide context for screen reader users as they are able to focus on the region.
+  // Do this by using table's <caption> to label the scrollable region using aira-labelleby
+  const attributescrollable = scrollable && {
+    className: 'ds-c-table__wrapper',
+    role: 'group',
+    'aria-labelledby': captionID.current,
+    'aria-live': 'polite',
+    'aria-relevant': 'additions',
+  };
+
+  const isTableCaptionComponents = (child) => {
+    return child && child.type === TableCaption;
+  };
+
+  const renderChildren = (captionId) => {
+    return React.Children.map(children, (child) => {
+      if (scrollable && isTableCaptionComponents(child)) {
+        // Extend props on tables before rendering.
+        return React.cloneElement(child, {
+          id: captionId,
+        });
+      }
+
+      return child;
+    });
+  };
+
+  const renderScrollableActiveAlert = () => {
+    const isActive = tabIndex === '0';
+    return scrollable && isActive ? scrollableActiveAlert : null;
+  };
+
+  return (
+    <>
+      {renderScrollableActiveAlert()}
+      <div ref={container} tabIndex={tabIndex} {...attributescrollable}>
+        <table className={classes} role="table" {...attributeOptions}>
+          {renderChildren(captionID.current)}
+        </table>
+      </div>
+    </>
+  );
+};
+
+Table.defaultProps = {
+  scrollableActiveAlert: (
+    <div className="ds-u-margin-y--1">
+      <Alert>
+        <p className="ds-c-alert__text" aria-hidden="true">
```

Comment:
Connected to the above feedback, if screen readers need to hear the alert on aria-live, we'll need to remove the `aria-hidden`.